### PR TITLE
issue-327: fix search input can not input chinese

### PR DESCRIPTION
**Pull request by wangliling-kkworld**
fix search input can not input chinese in react18 which use ReactDOM.createRoot to create app

### CODE REVIEW

File: src/KBarSearch.tsx [hunk 1]
```diff
@@ -26,6 +26,11 @@ export function KBarSearch(
     showing: state.visualState === VisualState.showing,
   }));
 
+  const [inputValue, setInputValue] = React.useState(search);
+  React.useEffect(() => {
```
Comment (timc1) [inline src/KBarSearch.tsx:30]: Curious how does setting the value as a side effect rather than directly fix the issue?

Comment (ashharrison90) [reply src/KBarSearch.tsx:30]: @timc1 i'm not the PR author, but i think it's less about "setting the value as a side effect" and more about "passing the updated value directly back to the input". now the value is passed straight back on line 60 instead of using the `search` value returned from `useKBar` which has to go through the pub/sub `useEffect` logic in `useStore`.

think both #327 and https://github.com/timc1/kbar/issues/341 are different manifestations of the same root cause - the `useEffect` in `useStore`.

Comment (timc1) [reply src/KBarSearch.tsx:30]: Yeah, my question is more so how that specifically fixes the issue of inputting different language characters?

Comment (Aqours) [reply src/KBarSearch.tsx:30]: React would trigger `onCompositionStart`, `onChange`, `onCompositionEnd` events in order when input CJK chars. This changes can trigger these events correctly, so inputing CJK chars is ok.

I test origianl code which only trigger `onCompositionStart`, `onChange` events on React@18, `onCompositionEnd` never be triggered(I don't know the specific reason).

**Review by timc1 — APPROVED** [review]

**Comment by ryuni-dev** [comment]
I still have the same problem in Korean...🥹

**Comment by Aqours** [comment]
> I still have the same problem in Korean...🥹

waiting [tailwind-nextjs-starter-blog](https://github.com/timlrx/tailwind-nextjs-starter-blog) project to update `kbar` dependency

File: src/KBarSearch.tsx (hunk 1 above)
Comment (AuSjukwa) [reply src/KBarSearch.tsx:30]: I found updating `KBarSearch` input value is going through the pub/sub useEffect in `useStore`, making the update priority low in concurrent mode, and react flush the update in async way. React runs the useEffect callback in  batchedUpdates context, there is an action called `restoreControlledState` is flushed before updating`KBarSearch` input value, the action will change the input DOM value into non-up-to-date value, and break the composition event.

In short words, flush updating `KBarSearch` input value is too late.
